cluster/src/main/java/com/hartwig/pipeline/input/ReduxFileLocator.java

```diff
@@ -43,9 +53,19 @@ private GoogleStorageLocation locateFile(SingleSampleRunMetadata metadata,
         var reference = metadata.type().equals(SingleSampleRunMetadata.SampleType.REFERENCE) ? metadata.sampleName() : "___";
         var pipelineRun = new PipelineRun(Pipeline.DNA_6_0, tumor, reference);
         var sampleType = metadata.type().equals(SingleSampleRunMetadata.SampleType.TUMOR) ? SampleType.TUMOR : SampleType.REFERENCE;
-        // We assume the default output format for pipeline5, both the BAM and the CRAM file are 2 levels below the pipeline output root.
-        var pipelineOutputLocation =
-                new PipelineOutputTemporaryLocation(URI.create(bamLocation).resolve("../../"), PipelineOutputStructure.PIPELINE5);
+
+        var bamTool = bamLocation.endsWith(".cram") ? Tool.CRAM : Tool.ALIGNER;
+        var relativeBamPath = PipelineFiles.get(pipelineRun,
+                        PipelineFiles.sampleTypeIs(sampleType),
+                        PipelineFiles.dataTypeIsAnyOf(DataType.ALIGNED_READS),
+                        PipelineFiles.toolIsAnyOf(bamTool))
+                .stream()
+                .findFirst()
+                .map(it -> it.getPathOrNull(inputBamDirectoryStructure))
+                .map(PipelineFilePath::toString)
+                .orElseThrow();
```

Comment:
Maybe pass a lambda to `orElseThrow` so you can throw an exception with a clear message? The default is a NoSuchElementException with the message "No value present", which will leave you wondering what's wrong.